tests/Fixer/Semicolon/SemicolonOnNewLineForChainedCallFixerTest.php

```diff
@@ -0,0 +1,288 @@
+                        ->method1("a", true) // this is a comment
+                        /* ->method2(true, false) */
+                        ->method3([1, 2, 3], ["a" => "b", "c" => 1, "d" => true])
+                        ->method4(1, "a", $f); /* this is a comment */
+                ?>',
+            ], [
+                '<?php
+                    $service->method1();
+                    $service->method2()->method3();
+                ?>',
+            ], [
+                '<?php
+                    $service->method1() ;
+                    $service->method2()->method3() ;
+                ?>',
+            ], [
+                '<?php
+
+                    $service
+                        ->method2(function ($a) {
+                            $a->otherCall()
+                                ->a()
+                                ->b()
+                            ;
+                        })
+                    ;
+                ?>',
+                '<?php
+
+                    $service
+                        ->method2(function ($a) {
+                            $a->otherCall()
+                                ->a()
+                                ->b()
+                            ;
+                        });
+                ?>',
+            ], [
+                '<?php
+
+                    $data = $service
+                        ->method2(function ($a) {
+                            $a->otherCall()
+                                ->a()
+                                ->b(array_merge([
+                                        1 => 1,
+                                        2 => 2,
+                                    ], $this->getOtherArray()
+                                ))
+                            ;
+                        })
+                    ;
+                ?>',
+                '<?php
+
+                    $data = $service
+                        ->method2(function ($a) {
+                            $a->otherCall()
+                                ->a()
+                                ->b(array_merge([
+                                        1 => 1,
+                                        2 => 2,
+                                    ], $this->getOtherArray()
+                                ));
+                        });
+                ?>',
+            ], [
+                '<?php
+
+                    $service
+                        ->method1(null, null, [
+                            null => null,
+                            1 => $data->getId() > 0,
+                        ])
+                        ->method2(4, Type::class)
+                    ;
+',
+                '<?php
+
+                    $service
+                        ->method1(null, null, [
+                            null => null,
+                            1 => $data->getId() > 0,
+                        ])
+                        ->method2(4, Type::class);
+',
+            ],
+        ];
+    }
+}
```

Comment:
test with non-standard whitespace config is missing.
grep for `testMessyWhitespaces` and add one here

Reply:
Thank you for reviewing! I've added the missing test, I hope it's correct now :)